docs/sources/mimir/configure/about-ingester-adaptive-limiters.md

```diff
@@ -0,0 +1,60 @@
+
+## Request rejection
+
+Ingesters support separate adaptive limiters for push and read requests since these may be overloaded by different resources at different times. When ingesters are heavily loaded these limiters may be full, and when the load is sustained, queues may fill up as well. When this happens, requests may be rejected, with a preference for rejecting read requests being before push requests.
+
+## Configure Grafana Mimir ingester adaptive limiters
+
+To enable Grafana Mimir ingester push and read adaptive limiters, set `-ingester.push-adaptive-limiter.enabled=true` and `-ingester.read-adaptive-limiter.enabled=true`. You can enable the adaptive limiters independently. Enabling one of them does not require enabling the other one.
+
+### Primary configuration
+
+When enabled, adaptive limiters come with some default configurations which should work well for many workloads. First among these are configurations for the short and long response time windows. These track how changes in recent, short-term response times compare to long-term response times.
+
+When requests are processed by ingesters, their response times are first aggregated in the short window. A quantile from these is regularly taken once the window is full and added to the long window. By default, the minimum duration of the short window is `1s`, the minimum number of responses that must be observed is `50`, and the aggregated response time quantile that's taken is `.9`. The long window defaults to `60` measurements, which are smoothed over time. These values can be changed via `short-window-min-duration`, `short-window-max-duration`, `short-window-min-samples`, `sample-quantile`, and `long-window`.
+
+The inflight request limit has a default min and max range from `2` to `200`, and an initial value of `20`. These can be changed via `min-inflight-limit`, `max-inflight-limit`, and `initial-inflight-limit`. Additionally, the inflight limit will ever only increase to a multiple of the current inflight requests, which defaults to `5.0`. This can be changed via `max-limit-factor`.
+
+### Additional configuration
+
+While response times are the primary mechanism for detecting overload, a secondary mechanism is tracking the correlation between inflight requests and throughput. The default size of this correlation window is `50` samples, which can be changed via `correlation-window`.
+
+The amount of queueing that is allowed when a limiter is full is based on the inflight request limit. By default, requests will begin to be rejected instead of queued when the queue reaches `2` times the inflight limit, and all requests will be rejected when the queue reaches `3` times the inflight limit. These can be configured via `initial-rejection-factor` and `max-rejection-factor`.
```

Comment:
```suggestion
The amount of queueing that is allowed when a limiter is full is based on the inflight request limit. By default, requests begin to be rejected instead of queued when the queue reaches `2` times the inflight limit, and all requests are rejected when the queue reaches `3` times the inflight limit. You can configure these settings via `initial-rejection-factor` and `max-rejection-factor`.
```